src/main/java/org/apache/commons/lang3/ArchUtils.java

```diff
@@ -53,94 +55,76 @@ private static final void init() {
      */
     private static final void init_IA64_64Bit() {
-        Processor ia64 = new Processor("ia64", ProcessorArch.BIT_64, ProcessorType.IA_64);
-        Processor ia64w = new Processor("ia64w", ProcessorArch.BIT_64, ProcessorType.IA_64);
-        addProcessors(ia64, ia64w);
+        Processor processor = new Processor(ProcessorArch.BIT_64, ProcessorType.IA_64);
+        addProcessors(Arrays.asList("ia64", "ia64w"), processor);
     }
 
     /**
      * Adding PPC 32 bit {@link Processor}'s to the map
      */
     private static final void init_PPC_32Bit() {
-        Processor ppc = new Processor("ppc", ProcessorArch.BIT_32, ProcessorType.PPC);
-        Processor power = new Processor("power", ProcessorArch.BIT_32, ProcessorType.PPC);
-        Processor powerpc = new Processor("powerpc", ProcessorArch.BIT_32, ProcessorType.PPC);
-        Processor power_pc = new Processor("power_pc", ProcessorArch.BIT_32, ProcessorType.PPC);
-        Processor power_rs = new Processor("power_rs", ProcessorArch.BIT_32, ProcessorType.PPC);
-        addProcessors(ppc, power, powerpc, power_pc, power_rs);
+        Processor processor = new Processor(ProcessorArch.BIT_32, ProcessorType.PPC);
+        addProcessors(Arrays.asList("ppc", "power", "powerpc", "power_pc", "power_rs"), processor);
     }
 
     /**
      * Adding PPC 64 bit {@link Processor}'s to the map
      */
     private static final void init_PPC_64Bit() {
-        Processor ppc64 = new Processor("ppc64", ProcessorArch.BIT_64, ProcessorType.PPC);
-        Processor power64 = new Processor("power64", ProcessorArch.BIT_64, ProcessorType.PPC);
-        Processor powerpc64 = new Processor("powerpc64", ProcessorArch.BIT_64, ProcessorType.PPC);
-        Processor power_pc64 = new Processor("power_pc64", ProcessorArch.BIT_64, ProcessorType.PPC);
-        Processor power_rs64 = new Processor("power_rs64", ProcessorArch.BIT_64, ProcessorType.PPC);
-        addProcessors(ppc64, power64, powerpc64, power_pc64, power_rs64);
+        Processor processor = new Processor(ProcessorArch.BIT_64, ProcessorType.PPC);
+        addProcessors(Arrays.asList("ppc64", "power64", "powerpc64", "power_pc64", "power_rs64"), processor);
     }
 
     /**
-     * Adds the given {@link Processor} to the map. The {@link Processor#getName()} is used as key
-     * and the {@link Processor} object as value.
+     * Adds the given {@link Processor} whith the given key {@link String} to the map.
      *
+     * @param key The key as {@link String}.
      * @param processor The {@link Processor} to add.
      * @throws UnsupportedOperationException When key already exists in map.
      */
-    private static final void addProcessor(Processor processor) throws UnsupportedOperationException {
-        if (!map.containsKey(processor.getName())) {
-            map.put(processor.getName(), processor);
+    private static final void addProcessor(String key, Processor processor) throws UnsupportedOperationException {
```

Comment:
If the parameters were changed to (Processor processor, String key ...) there would be no need to create the List and unpack it.